lambdas/service/openapi.json

```diff
@@ -683,23 +683,25 @@
                                                         "type": "boolean"
                                                     },
                                                     "plugins": {
-                                                        "type": "array",
-                                                        "items": {
-                                                            "type": "object",
-                                                            "properties": {
-                                                                "name": {
-                                                                    "type": "string"
+                                                        "type": "object",
+                                                        "properties": {
+                                                            "name": {
```

Comment:
This doesn't look right. There is no property called `name` in the response anymore. We're now using dynamic properties and the JSONSchema for that works differently. You can either use that or generate the schema. I think I prefer the latter.

Reply:
for reference here is the response from `index/catalogs` from my deployment.
```
{
  "default_catalog": "dcp2",
  "catalogs": {
    "dcp2": {
      "internal": false,
      "atlas": "hca",
      "plugins": {
        "repository": {
          "name": "tdr",
          "sources": [
            "tdr:broad-jade-dev-data:snapshot/hca_dev_20201203___20210524_lattice:42",
            "tdr:broad-jade-dev-data:snapshot/hca_dev_20210621_managedaccess_4298b4de92f34cbbbbfe5bc11b8c2422__20210622:"
          ]
        },
        "metadata": {
          "name": "hca"
        }
      }
    },
    "dcp2ebi": {
      "internal": false,
      "atlas": "hca",
      "plugins": {
        "repository": {
          "name": "tdr",
          "sources": [
            "tdr:broad-jade-dev-data:snapshot/hca_dev_20201023_ebiv4___20210302:4"
          ]
        },
        "metadata": {
          "name": "hca"
        }
      }
    },
    "lungmap": {
      "internal": false,
      "atlas": "lungmap",
      "plugins": {
        "repository": {
          "name": "tdr",
          "sources": [
            "tdr:broad-jade-dev-data:snapshot/lungmap_dev_20210412__20210414:"
          ]
        },
        "metadata": {
          "name": "hca"
        }
      }
    },
    "it2": {
      "internal": true,
      "atlas": "hca",
      "plugins": {
        "repository": {
          "name": "tdr",
          "sources": [
            "tdr:broad-jade-dev-data:snapshot/hca_dev_20201203___20210524_lattice:42",
            "tdr:broad-jade-dev-data:snapshot/hca_dev_20210621_managedaccess_4298b4de92f34cbbbbfe5bc11b8c2422__20210622:"
          ]
        },
        "metadata": {
          "name": "hca"
        }
      }
    },
    "it2ebi": {
      "internal": true,
      "atlas": "hca",
      "plugins": {
        "repository": {
          "name": "tdr",
          "sources": [
            "tdr:broad-jade-dev-data:snapshot/hca_dev_20201023_ebiv4___20210302:4"
          ]
        },
        "metadata": {
          "name": "hca"
        }
      }
    },
    "it3lungmap": {
      "internal": true,
      "atlas": "lungmap",
      "plugins": {
        "repository": {
          "name": "tdr",
          "sources": [
            "tdr:broad-jade-dev-data:snapshot/lungmap_dev_20210412__20210414:"
          ]
        },
        "metadata": {
          "name": "hca"
        }
      }
    }
  }
}
```

Reply:
openapi document:
```
{
  "default_catalog": "string",
  "catalogs": {
    "additionalProp1": {
      "atlas": "string",
      "internal": true,
      "plugins": {
        "repository": {
          "name": "string",
          "sources": [
            "string"
          ]
        },
        "metadata": {
          "name": "string"
        }
      }
    },
    "additionalProp2": {
      "atlas": "string",
      "internal": true,
      "plugins": {
        "repository": {
          "name": "string",
          "sources": [
            "string"
          ]
        },
        "metadata": {
          "name": "string"
        }
      }
    },
    "additionalProp3": {
      "atlas": "string",
      "internal": true,
      "plugins": {
        "repository": {
          "name": "string",
          "sources": [
            "string"
          ]
        },
        "metadata": {
          "name": "string"
        }
      }
    }
  }
}

```

Reply:
I do not understand the part of [your post](https://github.com/DataBiosphere/azul/pull/3109/files#r673377267) starting in

> openapi document:

It doesn't look like an OpenAPI specification document, but rather like an actual response document, but with fake values.

Reply:
Anyways, I want to change my mind about dynamically generating the schema. It creates a schema that is too big and duplicative. I think we should actually create a generic schema that allows dynamic properties, one that doesn't explicitly specify the plugin types. But the schema shouldn't be empty and just have `additionalProperties` set to True. IIRC, there is a way to restrict the values of each additional property to conform to a particular schema.

Reply:
> I do not understand the part of your post starting in

Ignore that comment please, it was not what i wanted to show, i copied that quickly off the openapi page, and did not realize it was an example response.